linera-execution/src/runtime.rs

```diff
@@ -68,6 +71,34 @@ struct ApplicationStatus {
     signer: Option<Owner>,
 }
 
+/// A loaded application instance.
+#[derive(Debug)]
+struct LoadedApplication<Instance> {
+    instance: Arc<Mutex<Instance>>,
+    parameters: Vec<u8>,
+}
+
+impl<Instance> LoadedApplication<Instance> {
+    /// Creates a new [`LoadedApplication`] entry from the `instance` and its `description`.
+    fn new(instance: Instance, description: UserApplicationDescription) -> Self {
```

Comment:
The expected argument would be the parameter. Not sure we need a (private) constructor TBH.

Reply:
I'm using this as a helper function in the `load_*_instance` methods. Instead of creating the `LoadedApplication` there which adds some noise to the code that don't mean much to those methods this allows them to use a single statement that better conveys my intention: creating a `LoadedApplication` from what `load_{contract,service}` returns.